File: server/src/main/java/gg/mineral/server/network/login/LoginAuthData.java
package gg.mineral.server.network.login;

import java.security.KeyPair;
import java.util.Random;

import gg.mineral.server.util.login.LoginUtil;
import lombok.Value;

@Value
public class LoginAuthData {
    private static final Random RANDOM = new Random();
    private KeyPair keyPair;
    private byte[] verifyToken = new byte[4];

    public LoginAuthData() {
        this.keyPair = LoginUtil.createKeyPair(1024);
        RANDOM.nextBytes(verifyToken);
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/login/LoginAuthData.kt
package gg.mineral.server.network.login

import gg.mineral.server.util.login.LoginUtil
import java.util.concurrent.ThreadLocalRandom

class LoginAuthData {
    val keyPair = LoginUtil.createKeyPair(1024)
    val verifyToken = ByteArray(4)

    init {
        ThreadLocalRandom.current().nextBytes(verifyToken)
    }
}
